### PR TITLE
fix freebsd build

### PR DESCRIPTION
**Pull request by majestrate**
this SHOULD fix the freebsd build that was broken by #689

### CODE REVIEW

File: llarp/util/threading.cpp [hunk 1]
```diff
@@ -16,14 +19,18 @@ namespace llarp
 #ifdef POSIX
 #ifdef __MACH__
       const int rc = pthread_setname_np(name.c_str());
+#else
+#ifdef __FreeBSD__
+      pthread_setname_np(pthread_self(), name.c_str());
```
Comment (michael-loki) [inline llarp/util/threading.cpp]: based on https://www.freebsd.org/cgi/man.cgi?query=pthread_setname_np&sektion=3&apropos=0&manpath=OSF1+V5.1%2falpha

this should be:

```c++
const int rc = pthread_setname_np(pthread_self(), name.c_str(), nullptr);
```

File: llarp/util/threading.cpp [hunk 2]
```diff
@@ -24,6 +30,8 @@ namespace llarp
         LogError("Failed to set thread name to ", name, " errno = ", rc,
                  " errstr = ", strerror(rc));
       }
+
+#endif
```
Comment (neuroscr) [inline llarp/util/threading.cpp]: is this a stray endif? why was it aadded?

Comment (michael-loki) [reply llarp/util/threading.cpp]: because of the `#else` on line 22.

File: llarp/util/threading.cpp [hunk 3]
```diff
@@ -16,6 +19,9 @@ namespace llarp
 #ifdef POSIX
 #ifdef __MACH__
       const int rc = pthread_setname_np(name.c_str());
+#else
+#ifdef __FreeBSD__
```
Comment (michael-loki) [inline llarp/util/threading.cpp]: `#elif defined(__FreeBSD__)`, and then that addresses @neuroscr’s comment

**Pull request merged**